shared/menubar/files-container.desktop.js

```diff
@@ -18,10 +30,21 @@ const mapDispatchToProps = dispatch => ({
 
 const mergeProps = (stateProps, dispatchProps) => ({
   onViewAll: dispatchProps.onViewAll,
-  tlfRows: stateProps._tlfRows.map(c => ({
-    onSelectPath: () => dispatchProps._onSelectPath(c.path),
-    path: FsTypes.pathToString(c.path),
-  })),
+  tlfRows: stateProps._tlfRows.map(c => {
+    const {participants, teamname} = FsUtil.tlfToParticipantsOrTeamname(FsTypes.pathToString(c.tlf))
```

Comment:
I'm confused about this suggestion. Why wouldn't we want to pass the data to the component in a presentable form? Guidance is to make the component as dumb as possible.

Reply:
It's redundant information passed around, and it's really just one utility function call, which I think can still count as dumb. It's not like the component has to deal with immutable data structures, etc.

I think this is a similar situation where we make menu item action handlers in container, but populate actual menu item text presentation in components.

Reply:
I think I'm going to keep it the way it is. The data needs to be calculated anyway, and it doesn't make a difference for performance whether it happens here or there. And it's cleaner to populate the component directly from a prop rather than generating it. Finally, `participants` and `teamname` can both be undefined, so we're saving not just a function call but an `|| []`/`|| ''` check.

Reply:
Performance is not my main concern. You have condition on `tlfType` in the component, and based on `tlfType` either use `participants` or `teamname`. `participants` and `teamname` have to be defined based on `tlfType` values, so the three things are tangled together with mental constraints that are not enforced by either flow or the language itself. So it's not clear from first look at the Props type how those would work with each other.

That being said, I think this is minor and I'm fine with the way you have now.

Reply:
I can probably refactor it to use `usernames.js`

Reply:
I'd feel bad if my comment prompted you to spend too much time on this, so please feel free to ignore this if it becomes complicated!